include/aspect/simulator.h

```diff
@@ -1223,6 +1226,13 @@ namespace aspect
        */
       std_cxx11::shared_ptr<MeltHandler<dim> > melt_handler;
 
+      /**
+       * Shared pointer for an instance of the NewtonHandler. This way,
+       * if we do not need the machinery for doing Newton stuff, we do
+       * not even allocate it.
+       */
+      std_cxx11::shared_ptr<NewtonHandler<dim> > newton_handler;
```

Comment:
remember to use `std_cxx11::unique_ptr` otherwise it will not compile for old compilers